ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,236 @@
+            span_id = trace_api.format_span_id(ctx.span_id)[2:]
+            span_name = "projects/{}/traces/{}/spans/{}".format(
+                self.project_id, trace_id, span_id
+            )
+
+            parent_id = None
+            if isinstance(span.parent, trace_api.Span):
+                parent_id = trace_api.format_span_id(span.parent.get_context().span_id)[2:]
+            elif isinstance(span.parent, trace_api.SpanContext):
+                parent_id = trace_api.format_span_id(span.parent.span_id)[2:]
+
+            start_time = get_time_from_ns(span.start_time)
+            end_time = get_time_from_ns(span.end_time)
+
+            attributes = _extract_attributes(span.attributes)
+
+            cloud_trace_spans.append({
+                "name": span_name,
+                "span_id": span_id,
+                "display_name": get_truncatable_str(span.name),
+                "start_time": start_time,
+                "end_time": end_time,
+                "parent_span_id": parent_id,
+                "attributes": attributes,
+                "links": _extract_links(span.links),
+                "status": _extract_status(span.status),
+                "time_events": _extract_events(span.events),
+            })
+
+        return cloud_trace_spans
+
+    def shutdown(self):
+        pass
+
+
+def get_time_from_ns(ns):
+    """Given epoch nanoseconds, split into epoch milliseconds and remaining nanoseconds"""
+    if not ns:
+        return None
+    return {'seconds': int(ns / 1e9), 'nanos': int(ns % 1e9)}
+
+
+def get_truncatable_str(str_to_convert, max_length=MAX_LENGTH):
+    """Truncate a string if exceed limit and record the truncated bytes
+        count.
+    """
+    truncated, truncated_byte_count = check_str_length(
+        str_to_convert, max_length
+    )
+
+    result = {
+        "value": truncated,
+        "truncated_byte_count": truncated_byte_count,
+    }
+    return result
+
+
+def check_str_length(str_to_check, limit=MAX_LENGTH):
+    """Check the length of a string. If exceeds limit, then truncate it.
+    """
+    str_bytes = str_to_check.encode("utf-8")
+    str_len = len(str_bytes)
+    truncated_byte_count = 0
+
+    if str_len > limit:
+        truncated_byte_count = str_len - limit
+        str_bytes = str_bytes[:limit]
+
+    result = str(str_bytes.decode("utf-8", errors="ignore"))
+
+    return result, truncated_byte_count
+
+
+def _extract_status(status: trace_api.Status):
+    """Convert a Status object to dict."""
+    if not status:
+        return None
+    status_json = {"details": None, "code": status.canonical_code.value}
+
+    if status.description is not None:
+        status_json["message"] = status.description
+
+    return status_json
+
+
+def _extract_links(links: Sequence[trace_api.Link]):
+    """Convert span.links"""
+    if not links:
+        return None
+    extracted_links = []
+    for link in links:
+        trace_id = trace_api.format_trace_id(link.context.trace_id)[2:]
```

Comment:
Following on the `[2:]` comment above, perhaps it would be good to put this in a helper function since it is used so often. Something like 
```
def _get_trace_id(context):
    # explanation for the [2:]
    return trace_api.format_trace_id(context.trace_id)[2:]
```

Same comment for the span id.